### PR TITLE
WIP integrating clinic finder frontend

### LINKED ISSUE #978: Integrate Clinic Finder with CM App
Thanks for creating an issue! Please fill out this form so we can be sure to have all the information we need, and to minimize back and forth.

* **What are we trying to do?**
Integrate Clinic Finder with the CM App so CM's can find the closest or cheapest clinic for their PT based on location and gestation limit. 

* **How could we solve this issue? (Not knowing is okay!)**
Needs UX exploration on the Abortion Info>Clinic Details Section

### PR DESCRIPTION
**Pull request by tingaloo**
I rule and have completed some work on Case Manager that's ready for review!

Tried to make a frontend similar to the wireframes posted in #978. Wanted to put this out and maybe @colinxfleming can see if it's in the right direction. I am mighty rusty on my bootstrap. 

The cool part about this so far is clicking the Find clinics > search clinics, and clicking find clinics again.

This pull request makes the following changes:
* Attempt to make a frontend

screen shot 2017-08-06 at 1 06 29 pm

It relates to the following issue #s: 
* Fixes #978

### CODE REVIEW

File: app/views/patients/_abortion_information.html.erb [hunk 1]
```diff
@@ -35,6 +37,16 @@
         <% end %>
       </div>
     </div>
+
+    <div class="row">
+      <div class="col-sm-12" id="clinic_finder_container">
```
Comment (colinxfleming) [inline app/views/patients/_abortion_information.html.erb]: Feeling this! Let's make it a gray background per @eheintzelman 's designs.

File: app/views/patients/_abortion_information.html.erb [hunk 2]
```diff
@@ -35,6 +37,16 @@
...
+
+        <%= link_to display_clinic_finder_path, class: 'display_button', remote: true do %>
```
Comment (colinxfleming) [inline app/views/patients/_abortion_information.html.erb]: IIRC in the designs, this was triggerable by a link/block/button-ish deal that would open -

screen shot 2017-08-05 at 6 20 42 pm

Bonus if we follow that design!

File: app/views/patients/_abortion_information.html.erb [hunk 3]
```diff
@@ -35,6 +37,16 @@
...
+        <h3>Find clinics</h3>
```
Comment (colinxfleming) [inline app/views/patients/_abortion_information.html.erb]: The pop open style from a call works well here, but if it winds up being easier we should feel free to just statically include it on page load and trigger it with a `hide`, and just render the `results` part. Might be easier to hook jquery into that way. But I'm definitely good either way as long as it works.

File: config/routes.rb [hunk 4]
```diff
@@ -4,13 +4,17 @@
     get 'dashboard', to: 'dashboards#index', as: 'dashboard'
     get 'reports', to: 'reports#index', as: 'reports'
     post 'search', to: 'dashboards#search', defaults: { format: :js }
+    get 'display_clinic_finder', to: 'patients#display_clinic_finder', as: 'display_clinic_finder'
+    get 'display_clinic_results', to: 'patients#display_clinic_results', as: 'display_clinic_results'
```
Comment (colinxfleming) [inline config/routes.rb]: We should hook this into `clinicfinders#search` if we can! I bet we can get away without a new endpoint for it.

File: app/views/clinicfinders/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+  <div class="col-sm-6">
+    <div class="info-form-left">
+      <div> Zip code </div>
```
Comment (colinxfleming) [inline app/views/clinicfinders/_form.html.erb]: If it's easier, you can use `bootstrap_form_for url: clinicfinder_path`, and that way you can take advantage of the very straightforward rails helpers that'll generate all the boostrap fluff for you. So something like:

```
<%= bootstrap_form_for url: clinicfinder_path do |f| %>
  <%= f.text_field :zip_code %>
  <%= f.select :approx_gestation_at_appt, weeks_options, label: 'Approx. gestation at appt' %>
<% end %>
```

Might have to put that in divs and rows and junk, but will definitely let you skip some of the scruff!

File: app/views/patients/_abortion_information.html.erb (hunk 3 above)
Comment (tingaloo) [inline app/views/patients/_abortion_information.html.erb]: I'm thinking of this from a testing perspective. If we can just do assert_texts, then this shouldn't be an issue. I will stick with the current solution until I find some other alternatives.

**Comment by tingaloo** [comment]
Progress: everything working on the frontend side. Awaiting the geocode backend to resolve on ```clinicfinders#search```
```E, [2017-08-06T13:03:22.575461 #14552] ERROR -- : Caught an error during Us geocoding call: Failed to open TCP connection to geocoder.us:80 (getaddrinfo: nodename nor servname provided, or not known)
E, [2017-08-06T13:03:22.575866 #14552] ERROR -- : /Users/tingaloo/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/net/http.rb:906:in `rescue in block in connect'
/Users/tingaloo/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/net/http.rb:903:in `block in connect'```

**Comment by colinxfleming** [comment]
Making a note that I'm doing work on the library in https://github.com/DCAFEngineering/clinic_finder/pull/21 to add medicaid and make the geocoding faster.

**Comment by colinxfleming** [comment]
Sorry, I know I'm the holdup on this! Doing some work soon to unstick.

**Comment by colinxfleming** [comment]
@tingaloo we can do some work on this in parallel -- pop open the gemfile and change the clinic_finder entry as follows:

```
gem 'clinic_finder', git: 'https://github.com/DCAFEngineering/clinic_finder.git', branch: 'object-orient'
```

and then use it in the controller as follows:

```
ClinicFinder::Locator.new(Clinic.all,
  gestational_age: params[:gestational_age], 
  naf_only: params[:naf_only], 
  medicaid_only: params[:medicaid_only]
).locate_nearest_clinics(params[:patient_zipcode])
```

Gestational age defaults to 999, naf_only defaults to false, and medicaid_only defaults to false. Let me know if you run into any snags, I'm going to keep working on this.

**Comment by colinxfleming** [comment]
(Price points aren't ready yet. Workin on it.)

**Comment by tingaloo** [comment]
When I search, I get an error.
```
NoMethodError - undefined method `<' for nil:NilClass:
   () Users/tingaloo/.rvm/gems/ruby-2.4.2/bundler/gems/clinic_finder-2fc4e9130198/lib/clinic_finder.rb:82:in `block in filter_by_params'
```

Gonna wait before I bring out the 🔨

**Comment by colinxfleming** [comment]
What are you submitting via the form?

**Comment by tingaloo** [comment]
Using 20001 for zip code and 4 weeks.

```
@abortron = ClinicFinder::Locator.new(Clinic.all,
  gestational_age: params[:gestational_age],
  naf_only: params[:naf_only],
  medicaid_only: params[:medicaid_only]
)

    @nearest = @abortron.locate_nearest_clinics(params[:patient_zipcode])
puts @nearest
```

**Comment by colinxfleming** [comment]
Closing in favor of #1287 , the public branch

**Pull request closed without merging**